32 bit Raspberry Pi

Trying to build natively on 32 bit Raspberry Pi OS:
```
pi@raspberrypi:~/smoldot $ cargo build --release
   Compiling smoldot v0.1.1 (/home/pi/smoldot)
error[E0308]: mismatched types
   --> src/database/full_sqlite.rs:612:65
    |
612 | ...                   .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
    |                                                   ^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 8 elements, found one with 4 elements

error[E0308]: mismatched types
   --> src/database/full_sqlite/open.rs:360:53
    |
360 |                         .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
    |                                                     ^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 8 elements, found one with 4 elements

error[E0308]: mismatched types
   --> src/database/full_sqlite/open.rs:380:57
    |
380 | ...                   .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
    |                                                   ^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 8 elements, found one with 4 elements

error[E0308]: mismatched types
   --> src/database/full_sqlite/open.rs:408:53
    |
408 |                         .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
    |                                                     ^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 8 elements, found one with 4 elements

For more information about this error, try `rustc --explain E0308`.
error: could not compile `smoldot` due to 4 previous errors
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,7 +21,7 @@ on:
   pull_request:
 
 jobs:
-  test:
+  test-64bits:
     name: Build and test
     runs-on: ubuntu-latest
     container:
@@ -153,7 +153,7 @@ jobs:
   all-ci:
     # This dummy job depends on all the mandatory checks. It succeeds if and only if CI is
     # considered successful.
-    needs: [test, wasm-node-check, check-features, check-rustdoc-links, fmt, clippy, cargo-deny, wasm-node-versions-match]
+    needs: [test-64bits, test-32bits, wasm-node-check, check-features, check-rustdoc-links, fmt, clippy, cargo-deny, wasm-node-versions-match]
     runs-on: ubuntu-latest
     steps:
      - run: echo Success

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,6 +32,18 @@ jobs:
     - name: Run tests, with all features
       run: cargo test --locked --workspace --all-features
 
+  test-32bits:
+    name: Build and test on 32bits platforms
+    runs-on: ubuntu-latest
+    container:
+      image: rust
+    steps:
+    - uses: actions/checkout@v2.3.4
+    - uses: Swatinem/rust-cache@v1
+    - run: rustup target add i686-unknown-linux-gnu
+    - name: Run tests, with all features
+      run: cargo test --target i686-unknown-linux-gnu --locked --workspace --all-features
+
   wasm-node-check:
     name: Check javascript node
     runs-on: ubuntu-latest

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -39,6 +39,7 @@ jobs:
       image: rust
     steps:
     - uses: actions/checkout@v2.3.4
+    - run: apt-get install libc6-dev:i386
     - uses: Swatinem/rust-cache@v1
     - run: rustup target add i686-unknown-linux-gnu
     - name: Run tests, with all features

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -39,7 +39,7 @@ jobs:
       image: rust
     steps:
     - uses: actions/checkout@v2.3.4
-    - run: apt-get install libc6-dev:i386
+    - run: apt-get update && apt-get install -y libc6-dev:i386
     - uses: Swatinem/rust-cache@v1
     - run: rustup target add i686-unknown-linux-gnu
     - name: Run tests, with all features

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -39,7 +39,7 @@ jobs:
       image: rust
     steps:
     - uses: actions/checkout@v2.3.4
-    - run: apt-get update && apt-get install -y libc6-dev:i386
+    - run: apt-get update && apt-get install -y libc6-dev-i386
     - uses: Swatinem/rust-cache@v1
     - run: rustup target add i686-unknown-linux-gnu
     - name: Run tests, with all features

--- a/src/database/full_sqlite.rs
+++ b/src/database/full_sqlite.rs
@@ -608,9 +608,7 @@ impl SqliteFullDatabase {
 
                             let mut statement = connection.prepare("INSERT INTO grandpa_triggered_authorities(idx, public_key, weight) VALUES(?, ?, ?)").unwrap();
                             for (index, item) in change.next_authorities.enumerate() {
-                                statement
-                                    .bind(1, i64::try_from(index).unwrap())
-                                    .unwrap();
+                                statement.bind(1, i64::try_from(index).unwrap()).unwrap();
                                 statement.bind(2, &item.public_key[..]).unwrap();
                                 statement
                                     .bind(3, i64::from_ne_bytes(item.weight.get().to_ne_bytes()))

--- a/src/database/full_sqlite.rs
+++ b/src/database/full_sqlite.rs
@@ -609,7 +609,7 @@ impl SqliteFullDatabase {
                             let mut statement = connection.prepare("INSERT INTO grandpa_triggered_authorities(idx, public_key, weight) VALUES(?, ?, ?)").unwrap();
                             for (index, item) in change.next_authorities.enumerate() {
                                 statement
-                                    .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
+                                    .bind(1, i64::try_from(index).unwrap())
                                     .unwrap();
                                 statement.bind(2, &item.public_key[..]).unwrap();
                                 statement

--- a/src/database/full_sqlite/open.rs
+++ b/src/database/full_sqlite/open.rs
@@ -357,7 +357,7 @@ impl DatabaseEmpty {
                     .unwrap();
                 for (index, item) in finalized_triggered_authorities.iter().enumerate() {
                     statement
-                        .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
+                        .bind(1, i64::try_from(index).unwrap())
                         .unwrap();
                     statement.bind(2, &item.public_key[..]).unwrap();
                     statement
@@ -377,7 +377,7 @@ impl DatabaseEmpty {
                         .unwrap();
                     for (index, item) in list.iter().enumerate() {
                         statement
-                            .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
+                            .bind(1, i64::try_from(index).unwrap())
                             .unwrap();
                         statement.bind(2, &item.public_key[..]).unwrap();
                         statement
@@ -405,7 +405,7 @@ impl DatabaseEmpty {
                     .unwrap();
                 for (index, item) in finalized_authorities_list.clone().enumerate() {
                     statement
-                        .bind(1, i64::from_ne_bytes(index.to_ne_bytes()))
+                        .bind(1, i64::try_from(index).unwrap())
                         .unwrap();
                     statement.bind(2, &item.public_key[..]).unwrap();
                     statement.next().unwrap();

--- a/src/database/full_sqlite/open.rs
+++ b/src/database/full_sqlite/open.rs
@@ -356,9 +356,7 @@ impl DatabaseEmpty {
                     .prepare("INSERT INTO grandpa_triggered_authorities(idx, public_key, weight) VALUES(?, ?, ?)")
                     .unwrap();
                 for (index, item) in finalized_triggered_authorities.iter().enumerate() {
-                    statement
-                        .bind(1, i64::try_from(index).unwrap())
-                        .unwrap();
+                    statement.bind(1, i64::try_from(index).unwrap()).unwrap();
                     statement.bind(2, &item.public_key[..]).unwrap();
                     statement
                         .bind(3, i64::from_ne_bytes(item.weight.get().to_ne_bytes()))
@@ -376,9 +374,7 @@ impl DatabaseEmpty {
                         .prepare("INSERT INTO grandpa_scheduled_authorities(idx, public_key, weight) VALUES(?, ?, ?)")
                         .unwrap();
                     for (index, item) in list.iter().enumerate() {
-                        statement
-                            .bind(1, i64::try_from(index).unwrap())
-                            .unwrap();
+                        statement.bind(1, i64::try_from(index).unwrap()).unwrap();
                         statement.bind(2, &item.public_key[..]).unwrap();
                         statement
                             .bind(3, i64::from_ne_bytes(item.weight.get().to_ne_bytes()))
@@ -404,9 +400,7 @@ impl DatabaseEmpty {
                     .prepare("INSERT INTO aura_finalized_authorities(idx, public_key) VALUES(?, ?)")
                     .unwrap();
                 for (index, item) in finalized_authorities_list.clone().enumerate() {
-                    statement
-                        .bind(1, i64::try_from(index).unwrap())
-                        .unwrap();
+                    statement.bind(1, i64::try_from(index).unwrap()).unwrap();
                     statement.bind(2, &item.public_key[..]).unwrap();
                     statement.next().unwrap();
                     statement.reset().unwrap();

